pkg/api/networkservice/mechanisms/wireguard/helpers.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wireguard
+
+import (
+	"net"
+	"strconv"
+
+	"github.com/pkg/errors"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+)
+
+// Mechanism - a wireguard mechanism utility wrapper
+type Mechanism interface {
+	// SrcIP -  src ip
+	SrcIP() net.IP
+	// DstIP - dst ip
+	DstIP() net.IP
+	// SrcPublicKey - source public key
+	SrcPublicKey() (string, error)
+	// DstPublicKey - destination public key
+	DstPublicKey() (string, error)
+	// SrcPort - Source interface listening port
+	SrcPort() (int, error)
+	// SrcPort - Destination interface listening port
+	DstPort() (int, error)
+}
+
+type mechanism struct {
+	*networkservice.Mechanism
+}
+
+// ToMechanism - convert unified mechanism to useful wrapper
+func ToMechanism(m *networkservice.Mechanism) Mechanism {
+	if m.Type == MECHANISM {
+		return &mechanism{
+			m,
+		}
+	}
+	return nil
+}
+
+func (m *mechanism) SrcIP() net.IP {
+	return net.ParseIP(m.GetParameters()[SrcIP])
+}
+
+func (m *mechanism) DstIP() net.IP {
+	return net.ParseIP(m.GetParameters()[DstIP])
+}
+
+func (m *mechanism) stringValue(parameter string) (string, error) {
```

Comment:
Would it make more sense to simply return the zero value rather than (string, error) if the parameter isn't found?

Reply:
I think the errors more a clear way of handling unexpected situations.

Reply:
If you look at the conventions used for example by the grpc stuff... its all 'Return zero value if there is no value there'... which is arguably how Golang thinks about things in general...

Reply:
I found that for using this convention we need to do refactoring in `mechanisms/common` mechanism. 
Can I solve this problem in separate PR which will apply this convention for all mechanisms?

Reply:
I'd rather *not* introduce more of the (string, error) pattern here for simple absence of a parameter.

Reply:
fixed for this mechanism, but they're a list of mechanism which should be refactored for this convention:
1. mechanisms/common
2. mechanisms/srv6
3. mechanisms/vxlan

Reply:
Could you open an issues for them?

Reply:
Yes, sure. At this moment github returns 500 err code on creating the issue. I'll try later.

Reply:
Done: https://github.com/networkservicemesh/api/issues/29